h2/src/main/org/h2/util/LocalDateTimeUtils.java

```diff
@@ -499,8 +499,15 @@ private static Object localDateFromDateValue(long dateValue)
         int year = DateTimeUtils.yearFromDateValue(dateValue);
         int month = DateTimeUtils.monthFromDateValue(dateValue);
         int day = DateTimeUtils.dayFromDateValue(dateValue);
-
-        return LOCAL_DATE_OF_YEAR_MONTH_DAY.invoke(null, year, month, day);
+        try {
+            return LOCAL_DATE_OF_YEAR_MONTH_DAY.invoke(null, year, month, day);
+        } catch (InvocationTargetException e) {
+            if (month == 2 && day == 29) {
+                // If proleptic Gregorian doesn't have such date use the next day
```

Comment:
since we're talking about proleptic, surely there should be a
  year < something
condition here?

Reply:
Such check is possible, but I don't see a clear reason for it. `GregorianCalendar` in Java can be configured to use proleptic Julian caledar, for exampe, and in this case we may store modern dates that don't exist in proleptic Gregorian.

JDBC allows to set dates with custom calendars and H2 supports this at least partially.

Reply:
I'm just not comfortable overriding default behaviour unless there is a clear bug in the underlyging JDK code. If people are passing in custom calendars, then those custom calendars should have no problem with weird dates like this. Or the people are passing in bad calendars, in which case they should be throwing an exception

Reply:
There are no way for `LocalDate` to use custom calendar. There is `ChronoDate` for it, but this type is not mentioned in JDBC specification at all.

Databases (including H2) seem to normalize such dates in plain `getDate()` calls without a provided custom `Calendar` instead of throwing an exception, so for symmetry we can normalize date in `getObject(..., LocalDate.class)` too.

Some databases, however, return something like `0000-00-00` for such dates.

Reply:
I pushed commit with additional check for Julian era, so only dates that can be set with default Java `Calendar` are converted if `LocalDate` does not support them.

You are right, if some software uses custom calendar to set some non-standard date it should use it to read such date.